### PR TITLE
[examples][frameworks] Add support for `vite`

### PR DESCRIPTION
**Pull request by StephDietz**
This PR adds automatic framework detection for Vite. It adds an example using Vue.js, but it should be noted that many popular javascript frameworks work seamlessly with Vite. The example could just as easily been React. 

https://vitejs.dev

### CODE REVIEW

File: packages/frameworks/src/frameworks.ts [hunk 1]
```diff
@@ -1576,11 +1543,47 @@ export const frameworks = [
...
+    },
+    settings: {
+      installCommand: {
+        placeholder: '`yarn install` or `npm install`',
+      },
+      buildCommand: {
+        placeholder: '`yarn build` or `npm run build`',
+      },
+      devCommand: {
+        placeholder: '`yarn dev` or `npm run dev`',
```
Comment (styfle) [inline packages/frameworks/src/frameworks.ts]: This needs to be the actual dev command

File: packages/frameworks/src/frameworks.ts [hunk 2]
```diff
@@ -1576,11 +1543,47 @@ export const frameworks = [
...
+            '"(dev)?(d|D)ependencies":\\s*{[^}]*"vite":\\s*".+?"[^}]*}',
+        },
+      ],
+    },
+    settings: {
+      installCommand: {
+        placeholder: '`yarn install` or `npm install`',
+      },
+      buildCommand: {
+        placeholder: '`yarn build` or `npm run build`',
```
Comment (styfle) [inline packages/frameworks/src/frameworks.ts]: This needs to be the actual build command

File: packages/frameworks/src/frameworks.ts [hunk 3]
```diff
@@ -1576,11 +1543,47 @@ export const frameworks = [
...
+      },
+      devCommand: {
+        placeholder: '`yarn dev` or `npm run dev`',
+      },
+      outputDirectory: {
+        value: 'dist',
+      },
+    },
+    devCommand: 'yarn dev',
```
Comment (styfle) [inline packages/frameworks/src/frameworks.ts]: This needs to be the actual dev command

File: packages/frameworks/src/frameworks.ts [hunk 4]
```diff
@@ -1576,11 +1543,47 @@ export const frameworks = [
...
+    buildCommand: 'yarn build',
```
Comment (styfle) [inline packages/frameworks/src/frameworks.ts]: This needs to be the actual build command

File: packages/frameworks/src/frameworks.ts (hunk 2 above)
Comment (leerob) [inline packages/frameworks/src/frameworks.ts]:
```suggestion
        placeholder: 'vite build',
```

File: packages/frameworks/src/frameworks.ts (hunk 1 above)
Comment (leerob) [inline packages/frameworks/src/frameworks.ts]:
```suggestion
        placeholder: 'vite',
```

File: packages/frameworks/src/frameworks.ts (hunk 3 above)
Comment (leerob) [inline packages/frameworks/src/frameworks.ts]:
```suggestion
    devCommand: 'vite',
```

File: packages/frameworks/src/frameworks.ts (hunk 4 above)
Comment (leerob) [inline packages/frameworks/src/frameworks.ts]:
```suggestion
    buildCommand: 'vite build',
```

Comment (leerob) [reply packages/frameworks/src/frameworks.ts]: Added based on https://vitejs.dev/guide/#command-line-interface

**Review by leerob — APPROVED** [review]

File: packages/frameworks/src/frameworks.ts [hunk 5]
```diff
@@ -1576,11 +1543,47 @@ export const frameworks = [
     getOutputDirName: async () => 'public',
     defaultVersion: '0.13.0',
   },
+  {
+    name: 'Vite',
+    slug: 'vite',
+    //demo: 'https://vite.examples.vercel.com',
```
Comment (styfle) [inline packages/frameworks/src/frameworks.ts]:
```suggestion
    demo: 'https://vite.examples.vercel.com',
```

File: packages/frameworks/src/frameworks.ts [hunk 6]
```diff
@@ -1576,11 +1543,47 @@ export const frameworks = [
...
+            '"(dev)?(d|D)ependencies":\\s*{[^}]*"vite":\\s*".+?"[^}]*}',
+        },
+      ],
+    },
+    settings: {
+      installCommand: {
+        placeholder: '`yarn install` or `npm install`',
+      },
+      buildCommand: {
+        placeholder: 'vite build',
```
Comment (styfle) [inline packages/frameworks/src/frameworks.ts]:
```suggestion
        placeholder: '`npm run build` or `vite build`',
```

File: packages/frameworks/src/frameworks.ts [hunk 7]
```diff
@@ -1576,11 +1543,47 @@ export const frameworks = [
...
+      },
+      devCommand: {
+        placeholder: 'vite',
+      },
+      outputDirectory: {
+        value: 'dist',
+      },
+    },
+    devCommand: 'vite',
```
Comment (styfle) [inline packages/frameworks/src/frameworks.ts:1580]:
```suggestion
    dependency: 'vite',
    devCommand: 'vite',
```

**Review by styfle — APPROVED** [review]

**Pull request merged**